src/Core/Services/Implementations/StripePaymentService.cs

```diff
@@ -762,16 +764,44 @@ private async Task ChangeOrganizationSponsorship(Organization org, OrganizationS
         var daysUntilDue = sub.DaysUntilDue;
         var chargeNow = collectionMethod == "charge_automatically";
         var updatedItemOptions = subscriptionUpdate.UpgradeItemsOptions(sub);
+        var isPm5864DollarThresholdEnabled = _featureService.IsEnabled(FeatureFlagKeys.PM5864DollarThreshold);
 
         var subUpdateOptions = new Stripe.SubscriptionUpdateOptions
         {
             Items = updatedItemOptions,
-            ProrationBehavior = "always_invoice",
+            ProrationBehavior = isPm5864DollarThresholdEnabled && invoiceNow
+            ? Constants.AlwaysInvoice
+            : Constants.CreateProrations,
             DaysUntilDue = daysUntilDue ?? 1,
             CollectionMethod = "send_invoice",
             ProrationDate = prorationDate,
         };
 
+        if (!invoiceNow && isPm5864DollarThresholdEnabled)
+        {
+            var upcomingInvoiceWithChanges = await _stripeAdapter.InvoiceUpcomingAsync(new UpcomingInvoiceOptions
+            {
+                Customer = storableSubscriber.GatewayCustomerId,
+                Subscription = storableSubscriber.GatewaySubscriptionId,
+                SubscriptionItems = ToInvoiceSubscriptionItemOptions(updatedItemOptions),
+                SubscriptionProrationBehavior = Constants.CreateProrations,
+                SubscriptionProrationDate = prorationDate,
+                SubscriptionBillingCycleAnchor = SubscriptionBillingCycleAnchor.Now
+            });
+
+            // also check to see if any of the plan IDs are for an annual plan, or if the update is for a plan upgrade
+            var immediatelyInvoice = upcomingInvoiceWithChanges.AmountRemaining >= 5000;
+
+            subUpdateOptions.BillingCycleAnchor = immediatelyInvoice
+                ? SubscriptionBillingCycleAnchor.Now
+                : SubscriptionBillingCycleAnchor.Unchanged;
+        }
+
+        if (invoiceNow && isPm5864DollarThresholdEnabled)
+        {
+            subUpdateOptions.BillingCycleAnchor = SubscriptionBillingCycleAnchor.Now;
```

Comment:
⛏️ This appears to be a duplicate of what you have on line `795`. Also, the "invoice now" behavior is handled already on `773` by setting the proration behavior to `always_invoice`, so there should be no need to add this block here.

Reply:
resolved